tools/cldr-apps/src/main/java/org/unicode/cldr/web/auth/GithubLoginFactory.java

```diff
@@ -0,0 +1,201 @@
+        GITHUB_CLIENT_SECRET = instance.getProperty("GITHUB_CLIENT_SECRET");
+        final File homeFile = ((CLDRConfigImpl) instance).getHomeFile();
+        final Path derPrivateFile = new File(homeFile, PRIVATE_FILE_NAME).toPath();
+        final Path derPublicFile = new File(homeFile, PUBLIC_FILE_NAME).toPath();
+        byte[] derPublic = null;
+        byte[] derPrivate = null;
+        try {
+            derPrivate = Files.readAllBytes(derPrivateFile);
+        } catch (java.io.IOException e) {
+            logger.log(
+                    java.util.logging.Level.SEVERE,
+                    "Could not load " + derPrivateFile + ": " + e.getMessage(),
+                    e);
+        }
+        try {
+            derPublic = Files.readAllBytes(derPublicFile);
+        } catch (java.io.IOException e) {
+            logger.log(
+                    java.util.logging.Level.SEVERE,
+                    "Could not load " + derPublicFile + ": " + e.getMessage(),
+                    e);
+        }
+
+        if (derPublic != null && derPrivate != null) {
+            try {
+                final KeyFactory rsaFactory = KeyFactory.getInstance("RSA");
+                publicKey = rsaFactory.generatePublic(new X509EncodedKeySpec(derPublic));
+                privateKey = rsaFactory.generatePrivate(new PKCS8EncodedKeySpec(derPrivate));
+            } catch (NoSuchAlgorithmException | InvalidKeySpecException ex) {
+                logger.log(
+                        java.util.logging.Level.SEVERE,
+                        "Could not read private/public keys: " + ex.getMessage(),
+                        ex);
+            }
+        }
+    }
+
+    @Override
+    public boolean valid() {
+        if (GITHUB_APP_ID == null
+                || GITHUB_APP_ID.isBlank()
+                || GITHUB_CLIENT_ID == null
+                || GITHUB_CLIENT_ID.isBlank()
+                || GITHUB_CLIENT_SECRET == null
+                || GITHUB_CLIENT_SECRET.isBlank()
+                || privateKey == null
+                || publicKey == null) {
+            return false;
+        }
+
+        return true;
+    }
+
+    // /** Construct the JWT identifying as the client */
+    // private String getClientJwt() {
+    //     final long now = System.currentTimeMillis();
+    //     return Jwts.builder()
+    //             .setSubject("CLDR SurveyTool")
+    //             .signWith(privateKey)
+    //             .setIssuedAt(new Date(now - (60 * 1000)))
+    //             .setExpiration(new Date(now + (10 * 60 * 1000)))
+    //             .setIssuer(GITHUB_CLIENT_ID)
+    //             .compact();
+    // }
+
+    @Override
+    public String getLoginUrl(LoginIntent intent) {
+        if (intent != LoginIntent.cla) {
+            throw new IllegalArgumentException("LoginIntent must be cla but got " + intent);
+        }
+        // Note: does not set redirect URI, that can be done on the front end.
+        // TODO: add 'state=…' to a temporary token.
```

Comment:
this TODO looks more serious

Reply:
It is a security improvement, but I think it could be in a follow up PR before going live as it does not affect the core functionality or UI

Reply:
Our application doesn't give the user any options or special privileges to be performed… The only thing they can do is temporarily associate their survey to account with GitHub and then the CLA is checked, so the GitHub login is not an attack Vector for elevating privileges

I can make another subtask to track.

Reply:
i'll just keep this issue open